### PR TITLE
DOCS-344 - CLI and Server Installation, Architecture

### PR DESCRIPTION
**Pull request by joel-hamill**
Internal staging server: http://staging-docs.confluent.io.s3-website-us-west-2.amazonaws.com/PR/770/4.1.0/ksql/docs/installation/index.html

### CODE REVIEW

**Comment by joel-hamill** [comment]
ping @rodesai for config file review: https://github.com/confluentinc/ksql/pull/974/commits/29ba8d94caa8f9ec22b75f380bca76bb6c48c39a

File: docs/installation/installing.rst [hunk 1]
```diff
@@ -1,4 +1,86 @@
...
 
 Installing KSQL
-===============
\ No newline at end of file
+===============
+
+KSQL is a component of |cp| and the KSQL binaries are located at `https://www.confluent.io/download/ <https://www.confluent.io/download/>`_
+as a part of the |cp| bundle.
+
+KSQL must have access to a running Kafka cluster, which can be on prem, |ccloud|, etc.
```
Comment (apurvam) [inline docs/installation/installing.rst]: nit: 'on prem' seems like internal jargon. maybe "which can be in your data center, in a public cloud, in |ccloud|, etc"

File: docs/installation/architecture.rst [hunk 2]
```diff
@@ -2,3 +2,10 @@
 
 KSQL Architecture
 -----------------
+
+KSQL is an abstraction of Kafka Streams API.
+
+KSQL consumes streams of data that are stored in Kafka topics. KSQL data can be structured as Avro, JSON, DELIMITED (CSV).
+
+There are KSQL servers, which form KSQL clusters, and there is a CLI client that connect to the KSQL clusters.
```
Comment (apurvam) [inline docs/installation/architecture.rst]: I thought we were moving away from the 'KSQL Cluster' terminology. Below we use 'KSQL server pools'. Perhaps it would be better to converge on that term.

File: docs/installation/installing.rst [hunk 3]
```diff
@@ -1,4 +1,86 @@
...
+    information, see :ref:`quickstart`.
+
+Follow these instructions to start KSQL server.
+
+#.  Customize the KSQL ``ksql-server.properties`` file.  By default, the configuration file is located at ``/etc/ksql/ksql-server.properties``.
+    The required parameters are ``bootstrap.servers`` and ``listeners``. You can also set any property the Kafka Streams
+    API or the Kafka producer and consumer would understand. For a description of common configurations, see :ref:`configuring-ksql`.
+
+    .. tip:: KSQL servers that share the same ``command`` topic belong to the same resource pool. By default, KSQL servers
+             use the ``ksql__commands`` command topic. To assign a server to a different pool, change the ``ksql.command.topic.suffix``
```
Comment (apurvam) [inline docs/installation/installing.rst]: This has changed. 

We no longer the the `ksql.command.topic.suffix` setting. The command topic name is determined by the `ksql.service.id`. The default service id is actually `ksql_`. Every internal topic will have a fixed prefix of `_confluent-ksql-`.

So the default command topic is actually `_confluent_ksql-ksql_`.

File: docs/installation/installing.rst [hunk 4]
```diff
@@ -1,4 +1,86 @@
...
+             setting. For example, if you change to ``ksql.command.topic.suffix = production_commands``, the command topic will be named
```
Comment (apurvam) [inline docs/installation/installing.rst]: If you set `ksql.service.id` to "production_deployment", then the ksql command topic will be `_confluent-ksql-production_deployment`.

File: docs/installation/installing.rst [hunk 5]
```diff
@@ -1,4 +1,86 @@
...
+             ``ksql__production_commands``.
+
+    Here are the default settings:
+
+    .. code:: bash
+
+        bootstrap.servers=localhost:9092
+        ksql.command.topic.suffix=commands
```
Comment (apurvam) [inline docs/installation/installing.rst]: As above, remove this.

File: docs/installation/server-config/index.rst [hunk 6]
```diff
@@ -1,172 +1,233 @@
...
     security
     ccloud-ksql
     avro-schema
 
-You can set the default properties for KSQL, :cp-javadoc:`Kafka’s Streams |streams/javadocs/index.html`, Kafka’s
-:cp-javadoc:`producer client |clients/javadocs/org/apache/kafka/clients/producer/ProducerConfig.html` and
-:cp-javadoc:`consumer client |clients/javadocs/org/apache/kafka/clients/consumer/ConsumerConfig.html`, and admin client properties in the ``/etc/ksql/ksql-server.properties`` file.
+KSQL configuration parameters can be set for Admin Client, KSQL Query, KSQL Server, Kafka Streams, and Kafka Client. The
+KSQL Server parameters are configured via the ``/etc/ksql/ksql-server.properties`` file. The KSQL CLI parameters are
+configured via the ``/etc/ksql/ksql-server.properties`` file or the SET statement in the KSQL CLI.
```
Comment (apurvam) [inline docs/installation/server-config/index.rst]: I think only the streams properties can be set with the `SET` command, not the server parameters.. the latter are set at startup and can only be in the properties file.

File: docs/installation/server-config/index.rst [hunk 7]
```diff
@@ -1,172 +1,233 @@
...
 
-You can set per-session properties for :cp-javadoc:`Kafka’s Streams |streams/javadocs/index.html`, Kafka’s
-:cp-javadoc:`producer client |clients/javadocs/org/apache/kafka/clients/producer/ProducerConfig.html` and
-:cp-javadoc:`consumer client |clients/javadocs/org/apache/kafka/clients/consumer/ConsumerConfig.html`, and admin client in
-the SET statement in the CLI.
+You can set the following parameters for the KSQL Server and KSQL CLI.
 
-To view the current configuration settings, use the ``SHOW PROPERTIES`` KSQL command.
+.. important:: KSQL Server configuration settings take precedence over those set in the KSQL CLI. For example, if a value
+               for ``ksql.server.id`` is set in both the KSQL Server and KSQL CLI, the KSQL Server value is used.
```
Comment (apurvam) [inline docs/installation/server-config/index.rst]: There is no such config as `ksql.server.id`. It is `ksql.service.id`.

File: docs/installation/server-config/index.rst [hunk 8]
```diff
@@ -1,172 +1,233 @@
...
-Here is the basic syntax.
+Admin Client
+    These configurations control the KSQL admin client and use the same parameters as Kafka Streams. These can be provided
+    via the properties file with the optional ``ksql.streams`` prefix.
 
-.. code:: java
-
-    <property-name>=<property-value>
+KSQL Server
+    These configurations control the general behavior of the KSQL Server. For example, ``ksql.command.topic.suffix`` and
```
Comment (apurvam) [inline docs/installation/server-config/index.rst]: As above, there is no `ksql.command.topic.suffix` anymore.

File: docs/installation/server-config/index.rst [hunk 9]
```diff
@@ -1,172 +1,233 @@
...
-        fail.on.deserialization.error=false
+    fail.on.deserialization.error=false
 
+^^^^^^^^^^^^^^^^^^^^^^^^^
 ksql.command.topic.suffix
-    The KSQL server command topic determines the resource pool. By default, KSQL servers use the ``ksql__commands`` command topic.
-    To assign a server to a different pool, change the ``ksql.command.topic.suffix`` setting. For example, if you change to ``'production_commands'``, in the ``ksql-server.properties`` file, the command topic will be named ``ksql__production_commands``.
+^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The KSQL Server command topic determines the resource pool. By default, KSQL Servers use the ``ksql__commands`` command topic.
```
Comment (apurvam) [inline docs/installation/server-config/index.rst]: As above, these defaults are now out dated and the `ksql.command.topic.suffix` doesn't exist. See the above comment for the correct content.

File: docs/installation/server-config/index.rst [hunk 10]
```diff
@@ -1,172 +1,233 @@
...
+To assign a server to a different pool, change the ``ksql.command.topic.suffix`` setting. For example, if you change to ``'production_commands'``, in the ``ksql-server.properties`` file, the command topic will be named ``ksql__production_commands``.
 
-    .. code:: java
+.. code:: java
 
-        ksql.command.topic.suffix=production_commands
+    ksql.command.topic.suffix=production_commands
```
Comment (apurvam) [inline docs/installation/server-config/index.rst]: Non existent config option.

File: docs/installation/installing.rst (hunk 3 above)
Comment (apurvam) [reply docs/installation/installing.rst]: This has changed. The default command topic now is `_confluent_ksql-default_command_topic`.

File: docs/installation/installing.rst (hunk 4 above)
Comment (apurvam) [reply docs/installation/installing.rst]: This has changed. The command topic will be `_confluent-ksql-production_deployment_command_topic`.

File: docs/installation/architecture.rst (hunk 2 above)
Comment (joel-hamill) [reply docs/installation/architecture.rst]: @miguno ^^

File: docs/installation/installing.rst (hunk 3 above)
Comment (joel-hamill) [reply docs/installation/installing.rst]: updated the text, and moved to the `/config-server/index.rst` topic.

File: docs/installation/server-config/index.rst [hunk 11]
```diff
@@ -172,6 +158,18 @@ ksql.schema.registry.url
 
 The Schema Registry URL path to connect KSQL to.
 
+.. _ksql-service-id:
+
+^^^^^^^^^^^^^^^
+ksql.service.id
+^^^^^^^^^^^^^^^
+
+The service ID of the KSQL server. By default, the service ID of KSQL servers is ``ksql_``. KSQL servers that share the same
```
Comment (miguno) [inline docs/installation/server-config/index.rst]: The new default value is `default_`, not `ksql_`.  There might be other places where we still refer to the outdated `ksql_`.

File: docs/installation/server-config/index.rst [hunk 12]
```diff
@@ -172,6 +158,18 @@ ksql.schema.registry.url
...
+``command`` topic belong to the same resource pool. Every internal topic has a fixed prefix of
```
Comment (miguno) [inline docs/installation/server-config/index.rst]: belong to the same resource pool -> belong to the same KSQL cluster

File: docs/installation/server-config/index.rst (hunk 11 above)
Comment (miguno) [reply docs/installation/server-config/index.rst]: I'd reword this whole section to:

> The service ID of the KSQL server. The default service ID is ``default_``. KSQL servers that share the same service id belong to the same KSQL cluster.
>
> Note: Every KSQL cluster has its own internal "command" topic, whose name is derived from ``ksql.service.id``. The default command topic name is ``_confluent-ksql-default__command_topic``.

Comment (joel-hamill) [reply docs/installation/server-config/index.rst]: @apurvam ^?

File: docs/installation/server-config/index.rst (hunk 6 above)
Comment (joel-hamill) [reply docs/installation/server-config/index.rst]: so the KSQL CLI can only modify Streams parms - and these can only be modified using the SET property...so `ksql-server.properties` is only applicable to KSQL server?

File: docs/installation/server-config/index.rst (hunk 11 above)
Comment (apurvam) [reply docs/installation/server-config/index.rst]: Michael's wording is accurate.

File: docs/installation/server-config/index.rst (hunk 6 above)
Comment (apurvam) [reply docs/installation/server-config/index.rst]: `ksql-server.properties` includes all properties: ksql server, streams, kafka clients, ksql engine,.

The properties you can set in the CLI are _only_ the streams. properties. If you pass the same `ksql-server.properties` file to the server and the CLI, the CLI will only effectively use the streams properties from that file.

File: docs/installation/server-config/index.rst (hunk 12 above)
Comment (apurvam) [reply docs/installation/server-config/index.rst]: I thought that we were using 'server pools' and not 'KSQL clusters' as a terminology. Did this change?

Most of the docs already refer to server pools.

Regardless, It would probably be more productive to fix this in a single pass audit after all the docs are merged.
